### PR TITLE
New module: Plink exclude

### PR DESCRIPTION
**Pull request by atrigila**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
 
This module allows to exclude variants (stored in a file) from a plink binary set.

### CODE REVIEW

File: modules/nf-core/plink/exclude/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if( "$bed" == "${prefix}.bed" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
+    """
+    plink \\
+        --bfile ${meta.id} \\
+        $args \\
+        --exclude $variants \\
+        --threads $task.cpus \\
+        --make-bed \\
```
Comment (nvnieuwk) [inline modules/nf-core/plink/exclude/main.nf:32]: Is `--make_bed` an argument that always has to be given when running this tool?

File: modules/nf-core/plink/exclude/meta.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: plink_exclude
+description: Exclude variant identifiers from plink bfiles
+keywords:
+  - exclude
+  - plink
+tools:
+  - plink:
+      description: Whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner.
+      homepage: https://www.cog-genomics.org/plink
+
```
Comment (nvnieuwk) [inline modules/nf-core/plink/exclude/meta.yml:10]:
```suggestion
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Some small comments :)

File: modules/nf-core/plink/exclude/main.nf (hunk 1 above)
Comment (atrigila) [reply modules/nf-core/plink/exclude/main.nf:32]: In fact, no. My first approach when designing this module was to pass it as an `ext.args`, following nf-core guidelines. However, I saw that `plink extract` has this argument explicit and I tried to be consistent between similar modules. As your input here is a bed-bim-fam and you would like to extract/exclude variants from this triset, the `--make-bed` forces the output to keep being a bed-bim-fam. I'd like your opinion on this, but I think that I'll be better to add the `--make-bed` to the `ext.args`.

Comment (nvnieuwk) [reply modules/nf-core/plink/exclude/main.nf:32]: I think you should keep it in `ext.args` and adjust your output channels to also work when this argument isn't given (I assume a BED file isn't created if this argument isn't given?)

Comment (atrigila) [reply modules/nf-core/plink/exclude/main.nf:32]: If you don't provide  `--make-bed` the program breaks with "`Warning: No output requested.  (Did you forget --make-bed?)  Exiting.` " `--make-bed`  makes the bed-bim-fam, not only the bed.

Comment (atrigila) [reply modules/nf-core/plink/exclude/main.nf:32]: Yes, you need to commonly pass  `--make-bed`  to get the output or the command `--recode` to transform the output to another file type. Thanks for your feedback! :)

Comment (nvnieuwk) [reply modules/nf-core/plink/exclude/main.nf:32]: Ah so there is an option to create another file type? Then you should adjust the module to work for both :)

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**